**HOMEWORK 3 - ASSIGNMENT 2**
_Ramda Yanurzha - ry692@nyu.edu_

In [31]:
__author__= 'ry692@nyu.edu'
import os
import sys
import numpy as np
import pylab as pl


###LAB QUESTION: 
NULL HYPOTHESIS: the % of former prisoners employed after release is the same or lower for candidates who participated in the program as for the control group, significance level p=0.05

In [73]:
alpha=0.05
P_0=0.035
P_1=0.701
n_0=409
n_1=564

#count
Nt_0=P_0*n_0
Nt_1=P_1*n_1


In [74]:
sp=(P_0*n_0+P_1*n_1)/(n_1+n_0)
print sp

0.421047276465


In [75]:
sp_stdev= lambda p, n: np.sqrt( p * ( 1 - p ) /n[0] +  p * ( 1 - p )/n[1]  )
sp_stdev_2y=sp_stdev((Nt_0+Nt_1)/(n_0+n_1),[n_0,n_1])
print P_0, n_0, n_1, sp_stdev_2y

0.035 409 564 0.0320658086057


In [76]:
zscore = lambda p0, p1, s : (p0-p1)/s
z_2y = zscore(P_1, P_0, sp_stdev_2y)
print z_2y

20.7697865408


In [77]:
##p-value for employment after 2 years
p_2y=1-0.9998

def report_result(p,a):
    print 'is the p value {0:.2f} smaller than the critical value {1:.2f}? '.format(p,a)
    if p<a:
        print "YES!"
    else: print "NO!"
    
    print 'the Null hypothesis is {}'.format( 'rejected' if p<a  else 'not rejected') 

    
report_result(p_2y,alpha)

is the p value 0.00 smaller than the critical value 0.05? 
YES!
the Null hypothesis is rejected


###HW3 Question 1
what if we used the values for where the former inmate was or was not ever employed in a CEO transitional job 3 years after release?

In [80]:
P_0=0.521
P_1=0.533
n_0=409
n_1=564
Nt_0=P_0*n_0
Nt_1=P_1*n_1

In [81]:
sp_stdev_3y=sp_stdev((Nt_0+Nt_1)/(n_0+n_1),[n_0,n_1])

In [82]:
z_3y = zscore(P_1, P_0, sp_stdev_3y)

In [83]:
print z_3y

0.370114343926


In [86]:
#P for z=0.749: 0.6736
Z_3y=0.6736
p_3y=1-Z_3y
report_result(p_3y,alpha)

is the p value 0.33 smaller than the critical value 0.05? 
NO!
the Null hypothesis is not rejected


###HW3 Question 2
Chi-square: employment for >6 consecutive months

|employed for >6 months |     employed   | not employed   |                   
|---------------------------|----------------|------------------|---------------------------|
| test sample               | $14.7*5.64$    | $85.3*5.64$      | 564                       |
| control sample            | $11.9*4.09$     | $88.1*4.09$      | 409                       |
|                           |                |                  |                           |
| total                     | 131.579        |  841.421        | 973         |

In [87]:
Ntot = 973
expected = 564*409*131.579*841.421
sample_values = [[14.7*5.64,85.3*5.64],[11.9*4.09,88.1*4.09]]
 
chisqstat= lambda N, values, expect : N*((values[0][0]*values[1][1]-values[0][1]*values[1][0])**2)/(expect)

print chisqstat(Ntot,  sample_values, expected)

1.58939333193


###Answer:
3.84 was mentioned on the lab because it corresponds to 0.05 probability of x^2 being bigger on 1-degree of freedom.
1.589 as the result of chi-square test implies that the null hypothesis is rejected because the corresponding probability (between 0.1 and 0.25) is higher than alpha (0.05).